modules/pulsar/pulsar.go

```diff
@@ -107,15 +108,16 @@ func WithTransactions() testcontainers.CustomizeRequestOption {
 	return func(req *testcontainers.GenericContainerRequest) {
 		WithPulsarEnv("transactionCoordinatorEnabled", "true")(req)
 
-		// add the waiting strategy for the transaction topic
-		defaultWaitStrategies.Strategies = append(
-			defaultWaitStrategies.Strategies,
+		// clone defaultWaitStrategies
+		ss := []wait.Strategy{
```

Comment:
I discovered a bug here in the case the tests were run sequentially, as the default strategies variable was overridden, causing the `with_transactions` test case to fail because it was expecting the wait strategy for `with_functions_workers`. Fixing it here to avoid the test failures that were discovered on this PR